### PR TITLE
integrated ODO analyze

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Signed-off-by: msivasubramaniaan [msivasub@redhat.com](mailto:msivasub@redhat.com)

ref: https://github.com/redhat-developer/vscode-openshift-tools/issues/2790

git-import-create-component-odo-analyze

**Devfile v1:**
git-import-create-component-odo-analyze-devfile-v1

### CODE REVIEW

File: src/webview/git-import/gitImportLoader.ts [hunk 1]
```diff
@@ -126,58 +88,59 @@ async function gitImportMessageListener(event: any): Promise<any> {
...
+                    return null;
                 }
-            } catch (error) {
-                vscode.window.showErrorMessage('Error occurred while killing the repository clone process');
-            }
-            panel?.dispose();
+                appendedUri = vscode.Uri.joinPath(workspacePath, event.projectName);
+                const isExistingComponent = await existingComponent(workspacePath);
+                if (isExistingComponent) {
+                    vscode.window.showErrorMessage(`Unable to create Component inside an existing Component: ${workspacePath}, Please select another folder`);
```
Comment (mohitsuman) [inline src/webview/git-import/gitImportLoader.ts:113]:
```suggestion
                    vscode.window.showErrorMessage(`Unable to create new component inside an existing component: ${workspacePath}. Please select another folder to proceed.`);
```

File: src/webview/git-import/gitImportLoader.ts [hunk 2]
```diff
@@ -237,6 +200,74 @@ export default class GitImportLoader {
...
 
+async function close(event: any) {
+    try {
+        const selctedFolder: vscode.Uri = vscode.Uri.from(event.folder);
+        if (childProcess && event.notification !== 'Component created') {
+            forceCancel = true;
+            treeKill(childProcess.pid, async () => {
+                await deleteDirectory(selctedFolder.fsPath);
+            });
+        } else if (event.notification === 'Component created') {
```
Comment (mohitsuman) [inline src/webview/git-import/gitImportLoader.ts:211]:
```suggestion
        } else if (event.notification === 'Component is successfully created.') {
```

File: src/webview/git-import/gitImportLoader.ts [hunk 3]
```diff
@@ -237,6 +200,74 @@ export default class GitImportLoader {
...
+            compDescription: compDescription,
+            parser: event.parser
+        });
+    } catch (error) {
+        if (error.message.toLowerCase().indexOf('failed to parse the devfile') !== -1) {
+            panel?.webview.postMessage({
+                action: 'devfileFailed'
+            });
+            const actions: Array<string> = ['Yes', 'Cancel'];
+            const devfileRegenerate = await vscode.window.showInformationMessage('Seems you are using Devfile v1. Will you be okay to regenerate it with v2?', ...actions);
```
Comment (mohitsuman) [inline src/webview/git-import/gitImportLoader.ts:245]:
```suggestion
            const devfileRegenerate = await vscode.window.showInformationMessage('We have detected that the repo contains congifuration based on devfile v1. The extension does not support devfile v1, will you be okay to regenerate a new devfile v2?', ...actions);
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
Few wording changes. I will look at the full workflow and recommend if that looks good. 

@msivasubramaniaan if you have a video for the workflow, that would be quick to review.

File: src/webview/git-import/gitImportLoader.ts (hunk 2 above)
Comment (msivasubramaniaan) [reply src/webview/git-import/gitImportLoader.ts:211]: Here notification not shown to the user. This is for internal purpose that the close event called after/during the component creation. https://github.com/redhat-developer/vscode-openshift-tools/blob/23cf2f124bd2b5c7ed216a44644954790d0d204b/src/webview/git-import/app/gitImport.tsx#L168

File: src/util/workspace.ts [hunk 4]
```diff
@@ -42,43 +42,45 @@ function createWorkspaceFolderItem(wsFolder: WorkspaceFolder) {
...
-    if (choice === AddWorkspaceFolder) {
-        const folders = await window.showOpenDialog({
+    if (choice === AddWorkspaceFolder || skipWindowPick) {
+        const selectedFolders = await window.showOpenDialog({
             canSelectFiles: false,
             canSelectFolders: true,
             canSelectMany: false,
             defaultUri: Uri.file(Platform.getUserHomePath()),
-            openLabel: 'Add context folder for component in workspace.',
+            openLabel: skipWindowPick ? 'Select as Repositroy Destination' : 'Add context folder for component in workspace.',
```
[the remark this answers is not in the document]
Comment (msivasubramaniaan) [reply src/util/workspace.ts:74]: VScode `git.clone` command used the pop up as **'Select as Repository Destination'**. So I used the same.
image

**'Select as Repository Destination'** means folder for clone the given URL

**Review by rgrunber — APPROVED** [review]
Change works for me. Definitely makes it easier having the recommendation. Feel free to squash + merge when ready.

**Comment by msivasubramaniaan** [comment]
> Few wording changes. I will look at the full workflow and recommend if that looks good.
> 
> @msivasubramaniaan if you have a video for the workflow, that would be quick to review.

Done

**Pull request merged**